.github/workflows/release.yaml

```diff
@@ -0,0 +1,38 @@
+name: Release to latest/edge
+
+on:
+  # TODO: figure out a release process for the charming-actions
+  workflow_dispatch:
+  # push:
+  #   branches:
+  #     - main
+  #   paths-ignore:
+  #     - ".github/**"
+  #     - ".gitignore"
+  #     - "tox.ini"
+
+jobs:
+  test:
+    uses: ./.github/workflows/build-and-test.yaml
+    secrets:
+      CHARMHUB_TOKEN: "${{ secrets.CHARMHUB_TOKEN }}"
+
+  release-to-charmhub:
+    name: Release to CharmHub
+    needs:
+      - test
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+        with:
+          fetch-depth: 0
+      - name: Select charmhub channel
+        uses: canonical/charming-actions/channel@631c2d944da2bd12430f1f3a954c8fffcf2385cd # 2.4.0
```

Comment:
Oh this one is really interesting and sounds like what we need to implement our release process, see https://github.com/canonical/charming-actions/tree/main/channel